index/scorch/segment/zap/docvalues.go

```diff
@@ -178,3 +179,24 @@ func (s *Segment) VisitDocumentFieldTerms(localDocNum uint64, fields []string,
 	}
 	return nil
 }
+
+// VisitableDocValueFields returns the list of fields with
+// persisted doc value terms ready to be visitable using the
+// VisitDocumentFieldTerms method.
+func (s *Segment) VisitableDocValueFields() ([]string, error) {
+	if len(s.fieldsInv) == 0 {
+		return nil, fmt.Errorf("VisitableDocValueFields: no fields" +
+			" for doc value persistence")
+	}
+
+	var rv []string
+	for fieldID, field := range s.fieldsInv {
+		if dvIter, ok := s.fieldDvIterMap[uint16(fieldID)]; ok {
```

Comment:
could write this one line instead of 2 if conditions as:
...; ok && dvIter != nil {